server/src/main/java/org/apache/druid/metadata/SQLMetadataSegmentManager.java

```diff
@@ -436,117 +466,94 @@ public ImmutableDruidDataSource getInventoryValue(String key)
   @Override
   public void poll()
   {
-    try {
-      ConcurrentHashMap<String, DruidDataSource> newDataSources = new ConcurrentHashMap<>();
-
-      log.debug("Starting polling of segment table");
+    synchronized (pollLock) {
```

Comment:
I don't see why we need an additional lock here, we change `startOrder` under `writeLock`, so at least on of the competing threads will have a local `startOrder` different while checking `startOrder == currentStartOrder` before calling `poll()`

Reply:
Added comment pointing that